smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolServiceClient.kt

```diff
@@ -175,66 +189,84 @@ open class HttpProtocolServiceClient(
+                writer.format("\$L: \$N = nil", it.name, it.type.toOptional())
+            }
         }
-        writer.openBlock(
-            "public convenience init(\$L) throws {",
-            "}",
-            propertyString,
-        ) {
-            writer.write(
-                "self.init(\$L)",
-                properties.joinToString(", ") {
+        writer.indent {
+            writer.openBlock("self.init(", ")") {
+                renderProperties(properties, true) {
                     if (it.default?.isAsync == true) {
                         it.name
                     } else {
                         it.default?.render(writer, it.name) ?: it.name
                     }
                 }
-            )
+            }
         }
+        writer.write("}")
         writer.write("")
     }
 
     private fun renderAsynchronousConfigInitializer(properties: List<ConfigProperty>) {
         if (properties.none { it.default?.isAsync == true }) return
 
-        writer.openBlock(
-            "public convenience init(\$L) async throws {", "}",
-            properties.joinToString(", ") { "${it.name}: ${it.type.toOptional().renderSwiftType(writer)} = nil" }
-        ) {
-            writer.write(
-                "self.init(\$L)",
-                properties.joinToString(", ") {
+        writer.openBlock("public convenience init(", ") async throws {") {
+            renderProperties(properties, true) {
+                writer.format("\$L: \$L = nil", it.name, it.type.toOptional().renderSwiftType(writer))
+            }
+        }
+        writer.indent {
+            writer.openBlock("self.init(", ")") {
+                renderProperties(properties, true) {
                     if (it.default?.isAsync == true) {
                         it.default.render(writer)
                     } else {
                         it.default?.render(writer, it.name) ?: it.name
                     }
                 }
-            )
+            }
         }
+        writer.write("}")
         writer.write("")
     }
+
     private fun renderServiceSpecificPlugins() {
         ctx.delegator.useFileWriter("Sources/${ctx.settings.moduleName}/Plugins.swift") { writer ->
             ctx.integrations
                 .flatMap { it.plugins(serviceConfig) }
                 .onEach { it.render(ctx, writer) }
         }
     }
+
+    // Renders the passed properties using the passed block to generate text,
+    // optionally comma-separating the lines as they are rendered.
+    protected fun renderProperties(
+        properties: List<ConfigProperty>,
+        commaSeparated: Boolean,
+        block: (ConfigProperty) -> String
+    ) {
+        val commaOrBlank = ",".takeIf { commaSeparated } ?: ""
+        properties.forEach { property ->
+            val separator = commaOrBlank.takeIf { property != properties.last() } ?: ""
+            writer.write("\$L\$L", block(property), separator)
+        }
+    }
```

Comment:
The method above is used to write the config params one per line, optionally separated by commas.

It's `protected` instead of `private` because it is used in a subclass in the SDK project.